homeassistant/components/ffmpeg/__init__.py

```diff
@@ -156,25 +161,25 @@ async def async_get_version(self):
         return self._version, self._major_version
 
     @property
-    def ffmpeg_stream_content_type(self):
+    def ffmpeg_stream_content_type(self) -> str:
         """Return HTTP content type for ffmpeg stream."""
         if self._major_version is not None and self._major_version > 3:
             return CONTENT_TYPE_MULTIPART.format("ffmpeg")
 
         return CONTENT_TYPE_MULTIPART.format("ffserver")
 
 
-class FFmpegBase(Entity):
+class FFmpegBase(Entity, Generic[_HAFFmpegT]):
     """Interface object for FFmpeg."""
 
     _attr_should_poll = False
 
-    def __init__(self, initial_state=True):
+    def __init__(self, ffmpeg: _HAFFmpegT, initial_state: bool = True) -> None:
```

Comment:
Changing the signature here looks like a breaking change for any custom integration that is using `FFmpegBase`

Reply:
Yeah it would be. Would you consider the `ffmpeg` integration public? I'm not sure about that. It isn't one of the core ones (`sensor`, `camera`, ...) but as you can see in this PR, I needed to adjust related (built-in) integrations as well.

Reply:
I'm not sure if we do consider it public, but I wouldn't be surprised if it causes someone a bit of a headache.  I think its fine to make the change, but we should probably tell custom devs in some way.

Reply:
Added a `Breaking change` section and the label. The alternative would be a dev blog post but that just feels way of the top for this one.